django_ratelimit/decorators.py

```diff
@@ -20,7 +23,9 @@ def _wrapped(request, *args, **kw):
                                          increment=True)
             request.limited = ratelimited or old_limited
             if ratelimited and block:
-                raise Ratelimited()
+                cls = getattr(
+                    settings, 'RATELIMIT_EXCEPTION_CLASS', Ratelimited)
+                raise (import_string(cls) if isinstance(cls, str) else cls)()
```

Comment:
Agree with doing this in general, as it makes testing far easier, but I also hear you on the performance concern. One option would be to move this into the top-level of the `decorator` function, which gets called far less often—though I admit I'm not positive that settings would _always_ be initialized already, depending when it's executed.

Reply:
I think making a global variable makes the most sense where the initial variable is set to `None`. On dev reload, a new thread is created by the web server, so a django setting "reload" would essentially be a new web server and thus the global variable will be reset to `None`.

The only concern would be when this package needs to support async caching and ORM. But that's a problem for another time.

I've also got classes tomorrow, so probably won't be able to get to this until next week.